src/main/webapp/threadHandler_test.js

```diff
@@ -0,0 +1,86 @@
+function hasValidCharacters(name) {
+  var regex = /^[a-zA-Z0-9-_]+$/;
+  return regex.test(name);
+}
+
+var assert = require('assert');
+
+describe('#hasValidCharacters()', function() {
+  const expectedOutput = true;
+  const lowerCase = "abcdefghijklmnopqrstuvwxyz";
+  const upperCase = lowerCase.toUpperCase();
+  const numbers = "1234567890";
+  const dashUnderscore = "-_";
+  const validCharacters = lowerCase+upperCase+numbers+dashUnderscore;
+  const invalidCharacters = "~!@#$%^&*?/()+=[]{}<>,.;\"\'\`\\\/\t\n\b\f\r";
+
+  describe('should return false when all characters are valid', function() {
+    it('should test lowercase letters are valid', function() {
+      let actualOutput = hasValidCharacters(lowerCase);
+      assert.equal(expectedOutput, actualOutput);
+    });
+    it('should test upercase letters are valid', function() {
+      let actualOutput = hasValidCharacters(upperCase);
+      assert.equal(expectedOutput, actualOutput);
+    });
+    it('should test numbers are valid', function() {
+      let actualOutput = hasValidCharacters(numbers);
+      assert.equal(expectedOutput, actualOutput);
+    });
+    it('should test dashes and underscores are valid', function() {
+      let actualOutput = hasValidCharacters(dashUnderscore);
+      assert.equal(expectedOutput, actualOutput);
+    });
+  });
+
+  describe('should return true when all characters are invalid', function() {
+    it('should test blank string is invalid', function() {
+      let input = "";
+      let actualOutput = hasValidCharacters(input);
+      assert.notEqual(expectedOutput, actualOutput);
+    });
+
+    [...invalidCharacters].forEach(function(c) {
+      it(`should test ${c} is invalid`, function() {
+        let actualOutput = hasValidCharacters(c);
+        assert.notEqual(expectedOutput, actualOutput);
+      });
+    });
+  });
+
+  describe('should check mixed strings correctly', function() {
+    let charTypes = [validCharacters, invalidCharacters];
```

Comment:
Your tests have a lot of logic in them. As a rule of thumb, tests should not have any loops or conditions because if that's the case then tests would need their own tests. It is ok to not test every case. Just make sure to test the cases that are most important.

Reply:
Same logic here as before, I'll pick a few inputs rather using logic, although it was fun generating random inputs 😄